api/query/search_test.go

```diff
@@ -104,3 +111,64 @@ func TestPrepareSearchResponse_qUC(t *testing.T) {
 func TestPrepareSearchResponse_pUC(t *testing.T) {
 	testIC(t, "/b/", false)
 }
+
+func TestStructuredSearchHandler_success(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	respBytes := []byte(`{}`)
+
+	server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "POST", r.Method)
+		assert.Equal(t, "/api/search/cache", r.URL.Path)
+		w.Write(respBytes)
+	}))
+
+	serverURL, err := url.Parse(server.URL)
+	assert.Nil(t, err)
+	hostname := serverURL.Host
+
+	api := sharedtest.NewMockAppEngineAPI(ctrl)
+
+	api.EXPECT().GetHostname().Return(hostname)
+	api.EXPECT().GetHTTPClient().DoAndReturn(func() *http.Client {
```

Comment:
Why not just `.Return(server.Client())` ?

Reply:
I was having problems and this `DoAndReturn()` was one of the things I tried to fix it. I can't remember whether or not the problem actually lay elsewhere. I'll change this to `Return()` if the test behaves correctly with it.